doc/architecture-specification.md

```diff
@@ -30,15 +30,47 @@ These keywords `SHALL` be in `monospace` for ease of identification.
 1. The user interface `SHALL` rely solely on the Aptos GraphQL endpoint for
    indexing purposes, enabling for example a leaderboard based on market
    capitalization of each emojicoin market.
+1. A `State` event `SHALL` be emitted for every operation, to enable fetching
+   market state as of the most recent event.
+1. Each market `SHALL` track a vector of internal `PeriodicState` event
+   accumulators, to enable volume and price versus times charts, with an
+   internal variable for the last time the event was emitted. Each granularity,
+   for example `1min`, `30min`, `1hr`, `6hr`, and `24hr`, `SHALL` track the last
+   time it was emitted, and use integer time truncation to check that a new
+   period has not started. When an operation results in a new period, a
+   `PeriodicState` event `SHALL` be emitted for each granularity that has
+   lapsed.
+1. A view function shall enable enable lookup of global total value locked,
+   number of emojicoin markets, and cumulative volume, via aggregators when
+   necessary.
+1. Events to simulate swap and liquidity provisioning operations `SHALL` be
+   calculated via simulation functions that can be called directly from view
+   functions.
+1. Simulation functions `SHALL` be paired with simulation commitment functions
+   such that results can be applied during runtime, using the same logic as
+   view function indexing functions.
```

Comment:
Is this essentially specifying that the underlying logic for the order flow and any effects it has on liquidity will be abstracted out to a view function? Just curious what this means here, trying to make sure I understand

Reply:
Yes, the idea is that the view function will rely on the same logic as the executor function.

It's similar to blockSTM in a prepare/commit scheme

Reply:
See here: https://github.com/econia-labs/emojicoin-dot-fun/blob/eb0e84fef21605f64429562eb2b1e99b91f77234/src/move/sources/emojicoin_dot_fun.move#L111